src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx

```diff
@@ -119,16 +117,14 @@ export class BlitzEvaluator implements WindowEvaluator {
 	}
 
 	private expectedBlitzes(window: HistoryEntry<EvaluatedAction[]>): number {
-		const priorCasts = this.pbCasts.filter(time => time <= window.start)
-		const priorRegens = this.rechargeTimes.filter(time =>
-			time <= (window.end ?? window.start) - END_OF_WINDOW_TOLERANCE
-		)
+		// We expect the player to burn off PB charges that were available when the window began, and any that became available with enough time before the window ended
+		const chargesBeforeWindow = (_.last(this.pbChargeHistory.filter(entry => entry.timestamp < window.start))?.current ?? this.maxCharges)
+		const chargesInWindow = this.pbChargeHistory.some(entry => entry.timestamp > window.start && entry.timestamp <= (window.end ?? this.pullEnd) - END_OF_WINDOW_TOLERANCE) ? 1 : 0
 
-		const castsAvailable = this.pbCharges - Math.min(this.pbCharges, priorCasts.length - priorRegens.length)
+		// If we entered the window with a PB already active and have enough time to use it before the window ends, expect that to get used too
+		const castsAvailable = chargesBeforeWindow + chargesInWindow + (this.perfectBalance.inBalance(window.start) && window.start + END_OF_WINDOW_TOLERANCE < (window.end ?? this.pullEnd) ? 1 : 0)
 
-		// Even though it is possible to have a triple blitz window
-		// we do not recommend it as a target since it is an advanced
-		// optimization that requires specific types of downtime
-		return Math.min(this.pbCharges, castsAvailable)
+		// If we're going into RoF with a blitz prepped, we can fit 3 blitzes in 11 GCDs (or at least one, if we're a short window at the end of the fight)
```

Comment:
Nit - can you confirm whether this comment is accurate? It says 3 but the `MAX_EXPECTED_BLITZES` const is 2.

Reply:
Three is possible in the case where a blitz is prepared ahead of time, but the logic to try to reflect that was bad. Added a specific const for that case and polished the returns to make it more readable and accurate.